openwisp_controller/pki/api/views.py

```diff
@@ -45,11 +49,15 @@ class CaDetailView(ProtectedAPIMixin, RetrieveUpdateDestroyAPIView):
     queryset = Ca.objects.all()
 
 
-class CrlDownloadMixinView(View, CrlDownloadMixin):
-    model = Ca
+class CrlDownloadView(ProtectedAPIMixin, RetrieveAPIView):
+    serializer_class = CaDetailSerializer
+    queryset = Ca.objects.none()
 
-    def get(self, request, pk):
-        return self.crl_view(request, pk)
+    def retrieve(self, request, *args, **kwargs):
+        instance = get_object_or_404(Ca, pk=kwargs['pk'])
+        return HttpResponse(
+            instance.crl, status=200, content_type='application/x-pem-file'
+        )
```

Comment:
@ManishShah120 have you made sure the CRL view from django-x509 is not used?

Reply:
Yes, I checked it. It's not being used.